### PR TITLE
RFC-0: Lightweight RFC Process

### PR DESCRIPTION
**Pull request by pivotal-amurmann**
This PR proposes a clearly defined decision making process for our community.

Note that I am following the process proposed in this PR itself.

I request that we arrive at a decision about adopting or rejecting this new process **by June 20th**. This should give us one week to read and discuss, which seems reasonable given the scope of the proposal.

### CODE REVIEW

**Comment by pivotal-jbarrett** [comment]
Can we title this PR "RFC-0: Lightweight RFC Process"? I like having the RFC number in the subject.

File: rfcs/000-rfc-lightweightRFCProcess.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Lightweight RFC Process
+
+**Geode-RFC-000**
```
Comment (pivotal-jbarrett) [inline rfcs/000-rfc-lightweightRFCProcess.md]: Can we not zero pad? I feel like that zero padding makes it less readable. IETF RFCs don't zero pad.

Comment (pivotal-jbarrett) [inline rfcs/000-rfc-lightweightRFCProcess.md]: Can we skip the dashes? Again I feel like the dashes effect readability. Perhaps just:
```
Geode RFC: 0
```

File: rfcs/000-rfc-lightweightRFCProcess.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+It’s encouraged to write an RFC for any major change. A major change might be:
+Addition of any major new feature or subsystem
+Changes that impact existing, public APIs. This includes Java APIs, but also things like GFSH commands and externally exposed protocols.
+
+### How to write an RFC?
+1. Copy the RFC template (in same folder as this document) and write your proposal! It's up to the author's discretion to decide which section in the template make sense for their proposal. It's recommended to cover the problem the proposal is solving, who it affects, how you’re proposing to solve it, and answers to frequently asked questions. Explicitly listing the goals will also make it easier to retrospect later on the proposal.
+2. Add your RFC to the geode/rfcs source directory, update to the next unique number. 
+3. Post a PR for your RFC putting [RFC] in the title. Make sure to state the deadline clearly in your PR.
```
Comment (pivotal-jbarrett) [inline rfcs/000-rfc-lightweightRFCProcess.md]: Can we change this to:
```
3. Post a PR for your RFC prefixing the title with `RFC-X: `, where X is the proposed RFC number. Make sure to state the deadline clearly in your PR. 
```

File: rfcs/000-rfc-lightweightRFCProcess.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+### When to write an RFC?
+Writing an RFC should be entirely voluntary. There is always the option of going straight to a pull request. However, for larger changes, it might be wise to de-risk the risk of rejection of the pull request by first gathering input from the community. Therefore it’s up to every member of our community to decide themselves when they want to reach for this tool.
+
+It’s encouraged to write an RFC for any major change. A major change might be:
+Addition of any major new feature or subsystem
+Changes that impact existing, public APIs. This includes Java APIs, but also things like GFSH commands and externally exposed protocols.
```
Comment (alb3rtobr) [inline rfcs/000-rfc-lightweightRFCProcess.md]: Add bullets to lines 50 & 51. In the preview of the file they are shown in the same paragraph than line 49.

File: rfcs/000-rfc-lightweightRFCProcess.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+2. Add your RFC to the geode/rfcs source directory, update to the next unique number. 
+3. Post a PR for your RFC putting [RFC] in the title. Make sure to state the deadline clearly in your PR. 
+The duration of the deadline is up to the author(s) and should depend on the size and complexity of the proposal. A reasonable time frame might be somewhere between a week and a month.
+4. Answer questions and concerns on the PR. Consider adding questions that get asked more than once to the FAQ section of the RFC.
+5. After the deadline for feedback has been reached summarize the consensus and your decision on the PR  thread. 
+    1. If the decision is to drop the proposal, the status should be updated to Dropped. 
+    2. If we are moving forward the status goes to *Active*
+    3. When there is a newer RFC that replaces this one the status goes to *Superseded* and the *Superseded By* gets updated with the number of the new RFC.
+    
+###Immutability & Errata
```
Comment (alb3rtobr) [inline rfcs/000-rfc-lightweightRFCProcess.md]: Add white space between the ### and the title. In the preview it is shown as a normal string, not as a title.

File: rfcs/000-rfc-lightweightRFCProcess.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+### Humble Advice
+Some things can be helpful to keep in mind when writing technical documents:
+
+1. Keep the document brief but complete. People don’t have time to thoroughly read and think about extremely long documents and they’ll receive less feedback compared with a shorter document. If you find it challenging to meet this limit then maybe the proposal is too big and could be broken up.
+2. Include evidence of the problem if at all possible, even if it’s anecdotal. This can help others see the core causes of the issue rather than only being able to comment on the diagnosis or solution. For example, consider linking to evidence inline, brief inline quotes, and/or footnotes.
+3. IETF RFCs you may see contain strict rules conveyed within the semantic meaning of *SHOULD*, *MUST*, and *MAY*. You don’t need to stress about the particulars of language or semantics when writing Geode-RFCs. Place importance explaining your problem and proposal clearly, succinctly, and convincingly over the specifics of implementation.
+
+## Alternatives
```
Comment (jdeppe-pivotal) [inline rfcs/000-rfc-lightweightRFCProcess.md]: Given that this is a PR for one style of RFC, this `Alternatives` section should be removed. If necessary, it should just be a comment to the PR.

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 1 above)
Comment (pivotal-amurmann) [reply rfcs/000-rfc-lightweightRFCProcess.md]: The dashes are markdown for making it bold. So it's a trade off between making the interpreted version more readable vs. the raw version. I am fine going either way.

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 5 above)
Comment (pivotal-amurmann) [reply rfcs/000-rfc-lightweightRFCProcess.md]: I am a little torn on this. You are absolutely right that this is really part of the discussion, rather than part of the thing that's being proposed and thus could live as a comment. I do see some value in preserving context from the discussion in the committed artifacts itself. That's why I also propose the FAQ section. While we will likely have access to the PR for a long time, we might lose that context if we ever move away from GitHub.

This brings up a related question: The PR discussion provides valuable context. It might be nice to make it easily discoverable. What do we think about linking to the PR from the header? It means an additional step when creating the PR because of the circular dependency, but likely will safe some searching in the future.

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 1 above)
Comment (pivotal-amurmann) [reply rfcs/000-rfc-lightweightRFCProcess.md]: Sounds good 👍🏻. Thoughts on keeping the file zero padded for sorting purposes?

**Comment by onichols-pivotal** [comment]
If we're going to use GitHub PR mechanism for RFCs, related commits should include `[skip ci]` to avoid running needless checks.

**Comment by metatype** [comment]
Most of our process-related information is captured in the wiki.  I'd rather see this information live there rather than in source control.

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 2 above)
Comment (metatype) [reply rfcs/000-rfc-lightweightRFCProcess.md]: I think we should use the wiki for proposals with a [DISCUSS] thread.  I don't see that a PR is helpful.

Comment (metatype) [reply rfcs/000-rfc-lightweightRFCProcess.md]: See https://cwiki.apache.org/confluence/display/GEODE/Project+Proposals+and+Specifications.

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 1 above)
Comment (pivotal-jbarrett) [reply rfcs/000-rfc-lightweightRFCProcess.md]: I would rather the file name match the title, no padding. Sorting by numerical order doesn't feel like it helps in finding what you want. The numbers are rather arbitrary to me.

Comment (pivotal-jbarrett) [reply rfcs/000-rfc-lightweightRFCProcess.md]: I don't mean the `**`, I mean the `-` used in the name. So change to this MD:
`**Geode RFC 0**`
should render:
**Geode RFC 0**

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 2 above)
Comment (pivotal-jbarrett) [reply rfcs/000-rfc-lightweightRFCProcess.md]: Can you please qualify how using the wiki is more helpful than the PR process?

The PR process:
* has bidirectional email integration
* has single place to have discussion
* supports inline commenting
* keeps RFC and errata changes documented with code changes in time
* keeps RFC with sources it effects
* Isn't another tool for contributors to use

The Wiki process:
* doesn't have group notification on changes, only followers are notified.
* discussions can happen in email or on the wiki and aren't integrated, so conversation can fragment on two systems, email and wiki.
* can't comment inline
* changes to wiki are tracked independent of code it may effect
* RFCs are not near the code, hiding away in wiki
* contributors have to get separate access and permissions edit wiki, is this possible for non-committers?

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 5 above)
Comment (pivotal-jbarrett) [reply rfcs/000-rfc-lightweightRFCProcess.md]: It might make more sense as a "prior art" section rather than alternatives.

**Comment by pivotal-jbarrett** [comment]
> If we're going to use GitHub PR mechanism for RFCs, related commits should include `[skip ci]` to avoid running needless checks.

Good catch, though we could change the CI to ignore this path.

**Comment by pivotal-jbarrett** [comment]
> Most of our process-related information is captured in the wiki. I'd rather see this information live there rather than in source control.

See https://github.com/apache/geode/pull/3706#discussion_r293550657

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 2 above)
Comment (upthewaterspout) [reply rfcs/000-rfc-lightweightRFCProcess.md]: A few corrections to the above - 

* The wiki can comment inline
* it has threaded discussions as well
* Discussions can be resolved.
* Contributors have access to the wiki
* Wiki changes and comments do send emails

I'll post a separate comment about why I think the wiki is a better place.

Comment (pivotal-jbarrett) [reply rfcs/000-rfc-lightweightRFCProcess.md]: Thanks for showing me the discussion thing. I didn't get any emails from your comments on the wiki though. Where is that email sent?

**Comment by upthewaterspout** [comment]
I think the wiki is a better place for proposals. Here's some advantages of the wiki:

1. All of our existing proposals and technical documentation is already there. Someone searching for information only has to search one place.
2. It easily supports embedding diagrams and other pictures into the proposal, which is important for complicated concepts.
3. You get to to review and comment inline the formatted text, not the plain text version
4. Finished RFCs are more easily discoverable - the wiki is indexed on google, and the geode space is searchable on the wiki without finding code as well.
5. Content from RFCs can be easily linked to or incorporated into our Architectural overview
6. Proposals that span multiple geode repositories have a reasonable place to stay.
7. The wiki is purpose built for technical documention. github is designed for storing code. Lots of little things (like how search results show up, how you add images to docs, how you link docs) work better on the wiki because that's what it is designed for.

**Comment by upthewaterspout** [comment]
> Would this process replace sending [PROPOSAL] emails to the dev list?

I  don't think it should. Most people aren't watching all of the github PRs, so they would never see RFCs in time. RFCs need more visibility, so they should be immediately sent to the mailing list and clearly labeled.

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 2 above)
Comment (upthewaterspout) [reply rfcs/000-rfc-lightweightRFCProcess.md]: I think you have to be watching the wiki page to get emails. You can see where your email is sent if you go to your profile in confluence.

Comment (pivotal-jbarrett) [reply rfcs/000-rfc-lightweightRFCProcess.md]: So here is my problem. If everyone on dev@geode doesn't get an email from the wiki inviting them to review and edit then you must send one manually. If you send one manually people will comment both via the email and via the wiki. Since that email won't feed back into the wiki, like a PR email does, we will have to either look both places for comments or remind the commenter to post those same comments on the wiki. 

When someone comments will it automatically make them a watcher of that document like PR does?

I worry a lot about this because I have seen how commenting about online documents have gone and they have always ended up split across the document and email and keeping all the threads intact is difficult. The priority of the initial RFC process is the C, comments. I believe we need to prioritize that aspect of it. After the commenting phase I really don't care where it ends up.

**Comment by pivotal-jbarrett** [comment]
> I think the wiki is a better place for proposals. Here's some advantages of the wiki:

See https://github.com/apache/geode/pull/3706#discussion_r293814190 for inline reply related to this topic.

**Comment by pivotal-jbarrett** [comment]
Maybe there is an experiment to be had here. What if RFC 0 is finished out with the PR process, then we immediately start RFC 1 to supersede RFC 0 to switch to the wiki. We could then conduct that experiment using the wiki, just as we are experimenting with the PR process here.

Effectively leave RFC 0 to discuss the everything but where the RFCs will be communicated, then resolve that, potentially, in a followup experiment.

**Comment by pivotal-jbarrett** [comment]
Using the PR process can those with effectively -1 votes or wanting changes discussed please use the "changes requested" approval state. This gives us a quick way to see who has issues with this PR to follow up with.

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 5 above)
Comment (pivotal-jbarrett) [reply rfcs/000-rfc-lightweightRFCProcess.md]: To elaborate, this section could list prior art and a rational for not using it. This way as others are reading the approved RFC later and wonder, "but why didn't we use X?", that perhaps X is referenced in prior art.

**Comment by metatype** [comment]
Regarding PR vs wiki, I don't think there's perfect solution.  As I mentioned earlier my *preference* is for wiki for some of the reasons Dan stated above.  Here's a few examples from Kafka:

https://lists.apache.org/thread.html/e1fef28f487ee7d52aa12b8312116b21194461aa6dc065b87910ef92@%3Cdev.kafka.apache.org%3E

https://lists.apache.org/thread.html/125bdd984fe0667962018da6ce10bce6d5895c5103955a8e4c730fef@%3Cdev.kafka.apache.org%3E

Note that the discussions happen solely via the dev@ list, not on the wiki.  Those things tend to show up on google:

https://www.google.com/search?q=geode+modularization

**Comment by metatype** [comment]
Another thing I like about the Kafka approach is that they have done synchronous, "in-person" meetings on some of the proposals, posted a link to the recording, and summarized the outcomes of the discussion to the dev list / wiki.

**Comment by pivotal-jbarrett** [comment]
@metatype 
> Note that the discussions happen solely via the dev@ list, not on the wiki. 

Can you follow the thread of conversation here though. We have had emails like this that are really hard to follow:
https://lists.apache.org/thread.html/f1e5af60522632c53c03f8c4940080c90ce8e889662a554e8324e03f@%3Cdev.kafka.apache.org%3E

**Comment by metatype** [comment]
I don't think there's a perfect solution for threaded communication about ideas.  I personally prefer email tooling for this, but that's just my preference.

(of course I prefer PR's for source code level discussions)

File: rfcs/000-rfc-lightweightRFCProcess.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+**Superseded by:** N/A
+
+**Related:** N/A
+
+
+## Problem
+With some of our recent efforts, we have seen consensus on design decisions take a very long time. In order to be able to keep moving forward as a successful project, we need to be able to get to consensus faster and ultimately get to execution faster. Therefore I propose a slightly more formal process for us to get to consensus.
+
+We already have gotten pretty good at sharing proposals with each other. Not much should need to change with that. It’s mainly the process around it that needs to be ironed out. This document will refer to that process as the “RFC process”.
```
Comment (balesh2) [inline rfcs/000-rfc-lightweightRFCProcess.md]: As this is the first mention of RFC, it would be nice to include what that stands for, to help out new contributors.

File: rfcs/000-rfc-lightweightRFCProcess.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+The main questions that the process needs to address are:
+What is enough consensus?
+How can we get to consensus in a reasonable time frame?
+
+## Solution
+The proposed solution to address the problem described above is to have an individual author or even better a group of authors submit a proposal to the community in order to gather feedback and achieve consensus. The RFC follows the same format as used by this proposal.
+
+Much inspiration for this proposal and in some cases whole segments have been drawn from Phil Calçado’s [Structured RFC Process](https://philcalcado.com/2018/11/19/a_structured_rfc_process.html). This proposal adopts Calçado’s work to work for an open source project like ours.
```
Comment (balesh2) [inline rfcs/000-rfc-lightweightRFCProcess.md]: The last sentence here is confusing. Perhaps change it to something like `This proposal adapts Calçado’s work for use in an open source project like ours.`

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 1 above)
Comment (pivotal-amurmann) [reply rfcs/000-rfc-lightweightRFCProcess.md]: I think the value of sorting by number might come once we are a few years in and have a fair number of superseded proposals.

**Comment by onichols-pivotal** [comment]
"if it didn't happen on the dev list, it didn't happen"

I'm not convinced that we need to impose any additional process here.  We already use the formalisms of time-boxed [DISCUSS] and [VOTE] threads via the dev list.  We [already](https://cwiki.apache.org/confluence/display/GEODE/Project+Proposals+and+Specifications) use the wiki to capture design documentation and evolve it from proposal to prototype to completed feature.  We already have GitHub and the PR process to review implementation details as the feature is realized.  

Apache values people over process.  It sounds like the problem trying to be solved here is "consensus on design decisions take a very long time".  Agreeing to use [Rough consensus](https://en.wikipedia.org/wiki/Rough_consensus) might be all that's needed to keep things moving forward.

That said, there are some great ideas in this proposal.  I would love to see them captured as "guidelines" at the top of https://cwiki.apache.org/confluence/display/GEODE/Project+Proposals+and+Specifications

**Comment by aaronlindsey** [comment]
I like this idea of using PRs instead of wiki/emails for RFCs because of the reasons already stated, as well as
- It's easy to see the voting status in the "Reviewers" side pane, as well as a voting summary at the bottom of the PR. With emails I have to scroll through every email and tally the votes in my head each time.
- It's easy to see when the RFC is still in the "voting" state by whether or not it is open/merged/closed. It's hard to know on the dev list when a consensus has been reached.
- With PRs, changes can only be accepted by Geode committers by merging the PR. Permissions on the wiki are more lax. For example, I have permission to edit the wiki but not permission to commit directly to Geode.

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 2 above)
Comment (upthewaterspout) [reply rfcs/000-rfc-lightweightRFCProcess.md]: I think the manual email to the dev list is a requirement for this RFC process. PRs don't email the dev list either, so few people are going to see the RFC unless they are watching all PRs and happen to catch that one of them is an RFC.

**Comment by upthewaterspout** [comment]
I do agree with Owen that this process should be a way of building consensus and getting feedback a larger change - not about voting an formally ratifying a document. I don't think we should be overly concerned with voting on these things at all - at the end of the day, sending an RFC is up to the committer to decide and following it afterwards is up to the committer as well.

On the wiki vs. PR thing - how about we just give people creating an RFC a choice of either for a little while, and then discuss which is working better. Once we do a few I think we will have more informed opinions.

**Comment by pivotal-amurmann** [comment]
I do very much like Jake's suggestion of seeing this RFC through as a PR. I also like Dan's idea of giving people a choice and see which one we like better. I do think though, that in that case we should make sure that the RFC also gets added to the wiki once approved. I am happy to do that for this PR.

**Comment by pivotal-jbarrett** [comment]
@upthewaterspout  This works!

alternative text

**Comment by metatype** [comment]
> * It's easy to see when the RFC is still in the "voting" state by whether or not it is open/merged/closed. It's hard to know on the dev list when a consensus has been reached.

I think that PR's for RFC's would *always* be merged regardless of "approved" / "rejected".  

> * With PRs, changes can only be accepted by Geode committers by merging the PR. Permissions on the wiki are more lax. For example, I have permission to edit the wiki but not permission to commit directly to Geode.

I'd like to see the RFC process be as inclusive as possible (let's not make it hard for people who aren't committers yet but want to contribute).

**Comment by pivotal-amurmann** [comment]
> Maybe there is an experiment to be had here. What if RFC 0 is finished out with the PR process, then we immediately start RFC 1 to supersede RFC 0 to switch to the wiki. We could then conduct that experiment using the wiki, just as we are experimenting with the PR process here.
> 
> Effectively leave RFC 0 to discuss the everything but where the RFCs will be communicated, then resolve that, potentially, in a followup experiment.

I really like this suggestion because it will allow us to make a more educated decision. Since the location question is still up in the air, I'd port this RFC to the wiki as well once it's finalized.
Is anyone in strong opposition to this?

File: rfcs/000-rfc-lightweightRFCProcess.md (hunk 2 above)
Comment (pivotal-amurmann) [reply rfcs/000-rfc-lightweightRFCProcess.md]: @pivotal-jbarrett I made the change that you requested that started this inline comment thread. Can you confirm that your original concern has been addressed?

**Comment by pivotal-amurmann** [comment]
When I proposed this RFC, I set expectations to wrap this up by today. In summary we seem to have largely found consensus on:
* What to cover in an RFC
* When to write and RFC
* How to write an RFC
* Associating a date with an RFC
* When to write and RFC

We do not have consensus on 
1. Should proposals live in the wiki or the code base?
2. Should discussion of a proposal happen
    1. In a PR
    2. In a [discuss] thread referencing a proposal in the wiki
    3. In the wiki itself

Given how many people actively work on this project I’d love to see a little wider participation on the location question we disagree on. 

 ## Summary of Location discussion
It seems that in the discussion so far we have been conflating where the discussion happens and where the proposals go afterwards. Jake, Dan and others have done a great job at listing lots of pros and cons of wiki vs. PR, let me try to split the arguments for the two different decisions we need to make:

## Where do we keep proposals?
### Proposals live in the wiki
1. All of our existing proposals and technical documentation is already there. Someone searching for information only has to search one place.
2. It easily supports embedding diagrams and other pictures into the proposal, which is important for complicated concepts.
3. Finished RFCs are more easily discoverable - the wiki is indexed on google, and the geode space is searchable on the wiki without finding code as well.
4. Content from RFCs can be easily linked to or incorporated into our Architectural overview
5. Proposals that span multiple geode repositories have a reasonable place to stay.
6. The wiki is purpose built for technical documentation. github is designed for storing code. Lots of little things (like how search results show up, how you add images to docs, how you link docs) work better on the wiki because that's what it is designed for.

### Proposals live in the code base
1. Keeps RFC and errata changes documented with code changes in time
2. Keeps RFC with sources it effects
3. With PRs, changes can only be accepted by Geode committers by merging the PR.

## Where do we discuss proposals?
### Discussion happens in the wiki
1. You get to to review and comment inline the formatted text, not the plain text version
2. Proposals that span multiple geode repositories have a reasonable place to stay.
3. Has threading of comments

### Discussion happens in a PR
1. Has bidirectional email integration
2. Has single place to have discussion
Discussion happens in the dev list (assuming doc lives in the wiki):
3. Has inline comments
4. It's easy to see the voting status in the "Reviewers" side pane, as well as a voting summary at the bottom of the PR.
5. It's easy to see when the RFC is still in the "voting" state by whether or not it is open/merged/closed

### Discussion happens only on the dev list
 1. Highly visible to all contributors.
 2. Very low barrier of entry.

(I left Discussion on dev list + gets submitted via PR out, since that combination makes no sense to me)

Let’s give this discussion a little bit more time till EoD Monday 6/24 and then wrap this up. If we cannot get a clear majority, I propose that we keep proposals in the wiki, since that’s the smallest change.

**Comment by pivotal-amurmann** [comment]
Given the discussion so far, my own vote is
1. Proposals live in the wiki
2. Proposals get discussed in the wiki

**Comment by pivotal-jbarrett** [comment]
@pivotal-amurmann 
Proposals live in the code base

>Proposals live in the code base
>1. Keeps RFC and errata changes documented with code changes in time
>2. Keeps RFC with sources it effects
>3. With PRs, changes can only be accepted by Geode committers by merging the PR.

You didn't include that as PR and living in the source that Google and other engines will also index it. There is a local site search as well. All the discussion is also searchable.

**Comment by pivotal-jbarrett** [comment]
I would like to point out that we have a recent proposal posted on the wiki that has fragmented discussions occurring in both email and inline commenting on the wiki. I you haven't "watched" the entire site or the specific page you have not seen the inline discussions.

**Comment by metatype** [comment]
> I would like to point out that we have a recent proposal posted on the wiki that has fragmented discussions occurring in both email and inline commenting on the wiki. I you haven't "watched" the entire site or the specific page you have not seen the inline discussions.

That's why Kafka requests that people comment via the mailing lists.

**Comment by pivotal-amurmann** [comment]
Closing in favor of wiki + [DISCUSS] thread. That's the smallest change for now and all the none-controversial parts of this proposal remain. We can always open another RFC if we want to discuss moving the discussion off the mailing list and the doc off the wiki.
